File: test files/order.php
<?php

/** Paid order status in PHP. */
const PAID_STATUS = 'paid';

class OrderPresenter
{
    /** Label shown for paid orders. */
    private string $statusLabel = 'Paid';

    /** Formats a PHP order status. */
    public function formatStatus(string $status): string
    {
        return $status === PAID_STATUS ? $this->statusLabel : 'Unknown';
    }
}

$presenter = new OrderPresenter();
$label = $presenter->formatStatus(PAID_STATUS);

File: test-fixtures/language-service/cpp/order.cpp
#include <string>

/** Formats the order status for customer-facing labels. */
std::string formatStatus(std::string status) {
  return status;
}

/// Presents order data in views.
class OrderPresenter {
 public:
  std::string label() {
    return formatStatus("paid");
  }
};

int main() {
  OrderPresenter presenter;
  return presenter.label().empty() ? 1 : 0;
}

